docs/docs/faq/index.md

```diff
@@ -10,4 +10,46 @@ version. Fetching an aggregate in a specific version directly from the `Aggregat
 first, but it is actually an anti-pattern in event sourcing. Basically what you're doing is a read action. This is why
 it does not belong to `AggregateRootRepository` but should be accomplished with
 [a projection or a read model](/docs/reacting-to-events/projections-and-read-models/). A read model does not have to be
-persistent, you can totally create an in-memory read model depending on your requirements.
\ No newline at end of file
+persistent, you can totally create an in-memory read model depending on your requirements.
+
+> How do I validate an aggregate against a service?
+
+The aggregate is tasked with maintaining the integrity of the model. Sometimes this requires validation of certain
+data against e.g. an external API or any other service. Aggregates, however, do not support constructor injection.
+This is because you are expected to pass all dependencies for a given "action" to the aggregate.
```

Comment:
Suggestion, replacing everything from "Aggregates, however," onwards:

Aggregates should not depend on external services for their construction. Still they are expected
to encapsulate the behaviour of interacting with said services. To enable this, services can be supplied to the
method that represents a particular interaction through. Unlike traditional method injection, these services are _not_
assigned to internal properties. The services are only used in the scope of executing the method.

Reply:
Thanks! Adjusted the text :-)

In this example, we always generate an event. In case I'm not interested in the invalid event and I require immediate consistency there, would it be okay to throw an exepction instead? Basically this:

```php
    public function changeIban(string $iban, IbanValidation $ibanValidation): self
    {
        if (!$ibanValidation->ibanIsValid($iban)) {
            throw new InvalidArgumentException(...);
        } else {
            $this->recordThat(new IbanWasChanged($iban));
        }
    }
```

Or does this violate some other ES principle?

Reply:
That's totally fine, emitting an event is primarily to setup for the next action. The failure paths are interesting for business analysts, but not a must.

Reply:
What do you think of 1ad4f252feb9631aa6e490301e9caa1b73b4218b then? :)
If that's okay for you, I think I'm done with this PR :)